gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/jupiter/http/vertx/VertxHttpServerRequest.java

```diff
@@ -224,10 +227,8 @@ public String host() {
     @Override
     public Maybe<Buffer> body() {
         // Reduce all the chunks to create a unique buffer containing all the content.
-        final Maybe<Buffer> body = chunks().reduce(Buffer::appendBuffer);
-        cacheChunks(body.toFlowable(), false);
-
-        return chunks.firstElement();
+        cacheChunks(chunks().reduce(Buffer::appendBuffer).toFlowable());
```

Comment:
in previous version, it wasn't actually working because we cached only the first time, and so sub call by policies resubscribe to the parent flowable.

Reply:
I have do some change on both Request and Reponse to share how they work, and also only caching when access is made on the body or chunks and not when setting it. Please have a new look.